src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -133,6 +135,20 @@ public static Index parseIndex(String oneBasedIndex) throws IllegalValueExceptio
         }
     }
 
+    //@@author sarahnzx
+    /**
+     * Checks if the specified social type is valid.
+     */
+    public static Optional<String> parseSelect(String arg) throws IllegalValueException {
+        requireNonNull(arg);
+        if (!("facebook".equals(arg) || "instagram".equals(arg))) {
```

Comment:
Can we use the identifiers and aliases in `SocialInfoMapping` instead of constants here? Maybe we can add a public method `getSocialType` to identify the social type based on input strings.